### PR TITLE
Common types refactor

### PR DESCRIPTION
**Pull request by aalavandhan**
* Factored out common types
* Optimized redemption amount calc

### CODE REVIEW

File: spot-contracts/contracts/_interfaces/CommonTypes.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (brandoniles) [inline spot-contracts/contracts/_interfaces/CommonTypes.sol:1]: This should be BUSL, right?

File: spot-contracts/contracts/_interfaces/IRolloverVault.sol [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (brandoniles) [inline spot-contracts/contracts/_interfaces/IRolloverVault.sol:1]: BUSL?

**Review by brandoniles — COMMENTED** [review]
Looks good, just the question about licensing

File: spot-contracts/contracts/_interfaces/CommonTypes.sol (hunk 1 above)
Comment (aalavandhan) [reply spot-contracts/contracts/_interfaces/CommonTypes.sol:1]: Right

File: spot-contracts/contracts/_interfaces/IRolloverVault.sol (hunk 2 above)
Comment (aalavandhan) [reply spot-contracts/contracts/_interfaces/IRolloverVault.sol:1]: I think GPL-3 for interfaces are fine

**Review by brandoniles — APPROVED** [review]
LGTM

**Pull request merged**